### PR TITLE
[Refactor](ReportHandler) Remove some unused schema_hash code in fe

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java [hunk 1]
```diff
@@ -296,32 +288,27 @@ private static void tabletReport(long backendId, Map<Long, TTablet> backendTable
             deleteFromMeta(tabletDeleteFromMeta, backendId, backendReportVersion);
         }
 
-        // 4. handle (be - meta)
-        if (foundTabletsWithValidSchema.size() != backendTablets.size()) {
-            deleteFromBackend(backendTablets, foundTabletsWithValidSchema, foundTabletsWithInvalidSchema, backendId);
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java:301]: Why delete this code? If it is removed, how to deal with the rubbish tablet on BE?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java [hunk 2]
```diff
@@ -140,104 +138,98 @@ public void tabletReport(long backendId, Map<Long, TTablet> backendTablets,
                         TTablet backendTablet = backendTablets.get(tabletId);
                         Replica replica = entry.getValue();
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java:139]: Add check here, check backendTablet.getTabletInfos size == 1;

File: fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java (hunk 1 above)
Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java:301]: my fault， I will fix it

File: fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java [hunk 3]
```diff
@@ -665,67 +662,42 @@ private static void deleteFromMeta(ListMultimap<Long, Long> tabletDeleteFromMeta
...
-        // This means that the meta of all backend tablets can be found in fe, we only need to process tablets with invalid Schema
-        if (foundTabletsWithValidSchema.size() + foundTabletsWithInvalidSchema.size() == backendTablets.size()) {
-            for (Long tabletId : foundTabletsWithInvalidSchema.keySet()) {
-                // this tablet is found in meta but with invalid schema hash. delete it.
-                int schemaHash = foundTabletsWithInvalidSchema.get(tabletId).getSchemaHash();
-                DropReplicaTask task = new DropReplicaTask(backendId, tabletId, schemaHash);
+        for (Long tabletId : backendTablets.keySet()) {
+            TTablet backendTablet = backendTablets.get(tabletId);
+            TTabletInfo backendTabletInfo = backendTablet.getTabletInfos().get(0);
+            boolean needDelete = false;
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java:673]: If tablet is not find in tablet meta, then the tablet is not deleted?

**Review by yiguolei — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 4]
```diff
@@ -1422,7 +1422,7 @@ private void handleShowTablet() throws AnalysisException {
                 }
                 dbName = db.getFullName();
                 Table table = db.getTableNullable(tableId);
-                if (table == null || !(table instanceof OlapTable)) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:1425]: No need to check null?

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:1425]: no need, null instanceof always return false

**Pull request merged**